packages/manager/modules/overthebox/src/overthebox/autoconfigure/dns/autoconfigure-dns.controller.js

```diff
@@ -0,0 +1,240 @@
+export default class OverTheBoxAutoconfigureDns {
+  /* @ngInject */
+  constructor($http, $q, $translate) {
+    this.$http = $http;
+    this.$q = $q;
+    this.$translate = $translate;
+  }
+
+  $onInit() {
+    this.displayAddServerButton = true;
+    this.displayAddLocalDomainButton = true;
+    this.showAddServer = false;
+    this.showAddLocalDomain = false;
+
+    this.nameserver = {
+      server: '',
+      priority: '',
+    };
+
+    this.localDomain = {
+      hostname: '',
+      ip: '',
+      priority: '',
+    };
+
+    return this.$q.all([this.loadLocalDomain(), this.loadNameserver()]);
+  }
+
+  loadLocalDomain() {
+    return this.$http
+      .get(`/overTheBox/${this.serviceName}/configuration/dns/localDomain`)
+      .then((response) => {
+        this.localDomains = response.data;
+        return response;
+      })
+      .catch((error) => {
+        // Display error message
+        this.errorMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_loading_failed',
+          {
+            errorMessage: error.data.message,
+          },
+        );
+      });
+  }
+
+  loadNameserver() {
+    return this.$http
+      .get(`/overTheBox/${this.serviceName}/configuration/dns/nameserver`)
```

Comment:
Consider to have a service file, to manage HTTP request.
More easy for search, add, update, replace and to migrate.